src/rospack.cpp

```diff
@@ -179,11 +179,8 @@ class Stackage
       assert(manifest_loaded_);
       // get name from package.xml instead of folder name
       tinyxml2::XMLElement* root = get_manifest_root(this);
-      for(tinyxml2::XMLElement* el = root->FirstChildElement("name"); el; el = el->NextSiblingElement("name"))
-      {
-        name_ = el->GetText();
-        break;
-      }
+      if(root->FirstChildElement("name"))
+        name_ = root->FirstChildElement("name")->GetText();
```

Comment:
Maybe extract the duplicate call into a local variable: `tinyxml2::XMLElement* el = root->FirstChildElement("name");`?

Reply:
sure!